src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -204,7 +204,8 @@ StorageObjectStorageQueue::StorageObjectStorageQueue(
     storage_metadata.setColumns(columns);
     storage_metadata.setConstraints(constraints_);
     storage_metadata.setComment(comment);
-    storage_metadata.settings_changes = engine_args->settings->ptr();
+    if (engine_args->settings)
+        storage_metadata.settings_changes = engine_args->settings->ptr();
```

Comment:
And then it will crash [here](https://github.com/ClickHouse/ClickHouse/blob/924e274cf263b9601bb32ff444679ab294585369/src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp#L623-L624)

Reply:
Turns out it will not crash because only MODFIFY and RESET SETTING are supported for this engine and there is 
https://github.com/ClickHouse/ClickHouse/blob/b3dd5431ba3b889ca1811a31edc14e1ca630aa34/src/Storages/AlterCommands.cpp#L1549-L1553
so I think I will add support for this in a separate PR (I mean ADD SETTING command)